packages/preact-urql/src/hooks/useQuery.ts

```diff
@@ -38,67 +50,120 @@ export type UseQueryResponse<Data = any, Variables = object> = [
   (opts?: Partial<OperationContext>) => void
 ];
 
+/** Convert the Source to a React Suspense source on demand */
+function toSuspenseSource<T>(source: Source<T>): Source<T> {
+  const shared = share(source);
+  let cache: T | void;
+  let resolve: (value: T) => void;
+
+  return sink => {
+    let hasSuspended = false;
+
+    pipe(
+      shared,
+      takeWhile(result => {
+        // The first result that is received will resolve the suspense
+        // promise after waiting for a microtick
+        if (cache === undefined) Promise.resolve(result).then(resolve);
+        cache = result;
+        return !hasSuspended;
+      })
+    )(sink);
+
+    // If we haven't got a previous result then start suspending
+    // otherwise issue the last known result immediately
+    if (cache !== undefined) {
+      const signal = [cache] as [T] & { tag: 1 };
+      signal.tag = 1;
+      sink(signal);
+    } else {
+      hasSuspended = true;
+      sink(0 /* End */);
```

Comment:
It really isn't great that we're manually sending source events here in case of breaking changes in Wonka (which are in fact coming) but I haven't really found a better implementation for this yet that's as short. The alternative is to use `Wonka.make` I suppose but then we need to manually pipe all events through.